rclpy/test/test_task.py

```diff
@@ -322,6 +322,34 @@ def cb(fut):
         f.add_done_callback(cb)
         assert called
 
+    def test_set_result_on_done_future_without_exception(self) -> None:
+        f = Future()
+        f.set_result(None)
+        self.assertTrue(f.done())
+        f.set_result(None)
+        self.assertTrue(f.done())
+
+    def test_set_result_on_cancelled_future_without_exception(self) -> None:
+        f = Future()
+        f.cancel()
+        self.assertTrue(f.cancelled())
+        f.set_result(None)
+        self.assertTrue(f.done())
+
+    def test_set_exception_on_done_future_without_exception(self) -> None:
+        f = Future()
+        f.set_result(None)
```

Comment:
Can we check the exception isn't there before and is after like this?
```suggestion
        f.set_result(None)
        self.assertIsNone(f.exception())
```